### PR TITLE
Various changes to the kernel

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: src/profile.c [hunk 1]
```diff
@@ -338,7 +335,6 @@ static inline void outputStat(Stat stat, int exec, int visited)
   // Statement not attached to a file
   if(nameid == 0)
   {
-    HashUnlock(&profileState);
     return;
   }
```
Comment (ChrisJefferson) [inline src/profile.c]: I think you also need to remove the Unlock at the end of this function (397)

Comment (fingolfin) [reply src/profile.c]: Damn, of course :/

**Comment by markuspf** [comment]
You somehow seem to have broken HPC-GAP.

**Comment by fingolfin** [comment]
Everything should be fine now, let's see if the tests pass.

**Comment by fingolfin** [comment]
@ChrisJefferson @markuspf merge?

**Pull request merged**